core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -0,0 +1,142 @@
+      currency.getCurrencyCode,
+      startTime,
+      endTime
+    )
+
+    SpendReportingResults(Seq(dailySpendAggregation), spendSummary)
+  }
+
+  case class BillingProjectSpendExport(billingProjectName: RawlsBillingProjectName, billingAccountId: RawlsBillingAccountName, spendExportTable: Option[String])
+
+  private def dateTimeToISODateString(dt: DateTime): String = dt.toString(ISODateTimeFormat.date())
+
+  private def getSpendExportConfiguration(billingProjectName: RawlsBillingProjectName): Future[BillingProjectSpendExport] = {
+    dataSource.inTransaction { dataAccess =>
+       dataAccess.rawlsBillingProjectQuery.load(billingProjectName)
+    }.map {
+      case Some(RawlsBillingProject(_, _, Some(billingAccount), _, _, _, _, _, Some(spendReportDataset), Some(spendReportTable), Some(spendReportDatasetGoogleProject))) =>
+        BillingProjectSpendExport(billingProjectName, billingAccount, Option(s"${spendReportDatasetGoogleProject.value}.${spendReportDataset.value}.${spendReportTable.value}"))
+      case Some(RawlsBillingProject(_, _, Some(billingAccount), _, _, _, _, _, _, _, _)) => BillingProjectSpendExport(billingProjectName, billingAccount, None)
+      case None => throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, s"project ${billingProjectName.value} not found"))
+      case _ => throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, s"billing account not found on billing project ${billingProjectName.value}"))
+    }
+  }
+
+  private def getWorkspaceGoogleProjects(billingProjectName: RawlsBillingProjectName): Future[Set[GoogleProject]] = {
+    dataSource.inTransaction { dataAccess =>
+        dataAccess.workspaceQuery.listWithBillingProject(billingProjectName)
+    }.map { workspaces =>
+      workspaces.collect {
+        case Workspace(_, _, _, _, _, _, _, _, _, _, WorkspaceVersions.V2, googleProjectId, _, _, _, _, _) => GoogleProject(googleProjectId.value)
+      }.toSet
+    }
+  }
+
+  private def validateReportParameters(startDate: DateTime, endDate: DateTime): Unit = {
+    if (startDate.isAfter(endDate)) throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, "start date must be before end date"))
+  }
+
+  def getSpendForBillingProject(billingProjectName: RawlsBillingProjectName, startDate: DateTime, endDate: DateTime): Future[Option[SpendReportingResults]] = {
+    validateReportParameters(startDate, endDate)
+    requireProjectAction(billingProjectName, SamBillingProjectActions.alterSpendReportConfiguration) { // todo: new action here? this is an okay approx. but could add a specific one
+      for {
+        spendExportConf <- getSpendExportConfiguration(billingProjectName)
+        workspaceProjects <- getWorkspaceGoogleProjects(billingProjectName)
+
+        query =
+        s"""
+           | SELECT
+           |  SUM(cost) as cost,
+           |  SUM(IFNULL((SELECT SUM(c.amount) FROM UNNEST(credits) c), 0)) as credits,
```

Comment:
@aherbst-broad I left in credits, but I don't understand what it is. If there's a value here does that mean the cost was covered by free credits and not billed to the user?

Reply:
Yeah, total billed = cost - credits (full disclosure these were pretty liberally lifted from [here](https://cloud.google.com/billing/docs/how-to/bq-examples)). That is, to match the top level item in the google cloud billing page, I believe we'd need to subtract credits from the cost.